### PR TITLE
Make a generic tier selector component

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

Build a generic version of the `TierSelector` component.

Refactor implementation a bit to consider accessibility; use "collapsible" instead of "accordion", use standard "radio group" instead of custom button implementation.

Build a specific version of the `TierSelector` component for the `ProductVariantSelector`.

## Justify why they are needed

We need to be able to reuse this component also for picking a deductible level.

We should aim to use standard form inputs over custom implementations.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1901** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1901" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1901?utm_source=stack-comment).

File: apps/store/src/components/ProductPage/PurchaseForm/ProductVariantSelector.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import styled from '@emotion/styled'
```
Comment (guilhermespopolin) [inline apps/store/src/components/ProductPage/PurchaseForm/ProductVariantSelector.tsx:1]: We already have a `ProductVariantSelector` component under `components` folder that's being used in PDPs. Maybe we should rename one of the components?!

image

**Review by guilhermespopolin — APPROVED** [review]

File: apps/store/src/components/ProductPage/PurchaseForm/ProductVariantSelector.tsx (hunk 1 above)
Comment (robinandeer) [reply apps/store/src/components/ProductPage/PurchaseForm/ProductVariantSelector.tsx:1]: ahh good point, let me see if I can come up with a better name 🤔

**Comment by robinandeer** [comment]
@robinandeer merged this pull request with [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1901).

**Pull request merged**